packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -98,7 +102,7 @@ abstract contract FundingRateApplier is Testable, Lockable {
         return FundingRateStoreInterface(finder.getImplementationAddress("FundingRateStore"));
     }
 
-    function _getLatestFundingRate() internal view returns (FixedPoint.Signed memory) {
+    function _getLatestFundingRate() internal returns (FixedPoint.Signed memory) {
```

Comment:
Can this cause a re-entrancy lockup?

1. This calls out to the FRS. (re-entrancy locked)
2. The FRS tries to distribute rewards.
3. The FRS calls into this contract to withdraw the fees. (re-entrancy guard tries to lock, but fails since it's already locked)

Reply:
Yeah good catch... this causes a reentrancy whenever a Perpetual contract interaction occurs while a pending funding rate proposal has expired.

1. The Perpetual contract asks the FRS for `getFundingRateForContract()`
2. The FRS then sees a pending proposal has expired and attempts to disperse proposer rewards via `perpetual.withdrawFundingRateFees()`. This re-enters the perpetual contract.

I propose that we convert `getFundingRateForContract()` into a view method that does NOT attempt to disperse proposer rewards.